project/components/crud.py

```diff
@@ -47,6 +48,10 @@ class CRUD:
 
     session: AsyncSession
     model: Type[DBModel]
+    db_error_codes: dict[str:Exception] = {
```

Comment:
Please change here to `ServiceException` since we know what kind of exceptions we expect here.
```suggestion
    db_error_codes: dict[str, ServiceException] = {
```